src/components/Header.tsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+
+type HeaderProps = {
+  inputRef: React.Ref<HTMLInputElement>;
```

Comment:
The ref should be passed to child component by using `forwardRef`
`const MyIChildComponent= forwardRef(function MyInput(props, ref)`

here is documentation https://react.dev/reference/react/forwardRef

Reply:
![image](https://github.com/user-attachments/assets/4075fdab-0bf4-41bc-9f79-7c8204ddc362)